packages/core/assistant/src/session/session.ts

```diff
@@ -180,6 +152,7 @@ export class AiSession {
         yield* this.messageQueue.offer(toolResultMessage);
       } while (true);
 
+      // TDOO(burdon): Comment required.
```

Comment:
the queues shutting down will signal to stream consumers that the session has completed and no more messages are coming